src/common/Error/Error.ts

```diff
@@ -0,0 +1,121 @@
+import {ExtensionName} from '../../Models/Interfaces/Api';
+
+// User Error
+/**
+ * Error class used when user cancel operation.
+ */
+export class OperationCanceledError extends Error {
+  constructor(message: string) {
+    super(`Operation cancelled: ${message}`);
+    this.name = 'CancelOperationError';
+  }
+}
+
```

Comment:
Do we want to define SystemError class and UserError class and put either of two as base class for all these error classes. For UserError class, we can add one additional field like "suggestion". It will be simpler for any caller catching errors and treat them differently for user/system errors.